File: adge/lib/models/calendario.dart
import 'dart:convert';

import 'package:adge/models/evento.dart';

class Calendario {
  Calendario({
    required this.idCalendario,
    required this.evento,
    required this.fechaInicio,
    required this.fechaFin,
  });

  int idCalendario;
  Evento evento;
  DateTime? fechaInicio;
  DateTime? fechaFin;

  factory Calendario.fromJson(String str) =>
      Calendario.fromMap(json.decode(str));

  String toJson() => json.encode(toMap());

  factory Calendario.fromMap(Map<String, dynamic> json) => Calendario(
        idCalendario: json["idCalendario"],
        evento: Evento.fromMap(json['evento']),
        fechaInicio: DateTime.parse(json["fechaInicio"]),
        fechaFin: DateTime.parse(json["fechaFin"]),
      );

  Map<String, dynamic> toMap() => {
        "idCalendario": idCalendario,
        "evento": evento.toMap(),
        "fechaInicio": fechaInicio,
        "fechaFin": fechaFin
      };
}

File: adge_back_end/Adge.Model/Calendario/Calendario.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Adge.Model
{
    public class Calendario
    {
        public int idCalendario { get; set; }
        public Evento evento { get; set; }
        public DateTime fechaInicio { get; set; }
        public DateTime fechaFin { get; set; }
    }

    public class CalendarioPgo
    {
        public String idCalendario { get; set; }
        public String idEvento { get; set; }
        public String fechaInicio { get; set; }
        public String fechaFin { get; set; }
    }
}
